app/controllers/decks_controller.rb

```diff
@@ -1,13 +1,124 @@
 class DecksController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
+  rescue_from ActiveRecord::RecordInvalid, with: :invalid_record
```

Comment:
You could almost put those in the `ApplicationController` instead of all the individual subclasses.